tensorflow-tools/src/main/java/org/tensorflow/tools/Shape.java

```diff
@@ -61,6 +65,15 @@ public static Shape of(long... dimensionSizes) {
     return new Shape(dimensionSizes);
   }
 
+  public static Shape of(long firstDimensionSize, long[] otherDimensionSizes) {
+    long[] dimensionSizes = new long[otherDimensionSizes.length + 1];
+    dimensionSizes[0] = firstDimensionSize;
+    System.arraycopy(
+        otherDimensionSizes, 0, dimensionSizes, 1, otherDimensionSizes.length);
+
+    return Shape.of(dimensionSizes);
+  }
```

Comment:
I'm using it in `Dataset.batch`. It's just a bit of a convenience so that you can append to the beginning of an existing Shape. 

I currently have this statement:
```
List<Shape> batchOutputShapes =
        outputShapes.stream().map(s -> Shape.of(-1, s.asArray())).collect(Collectors.toList());
```

which we could replace with something like this, if we remove the utility:
```
List<Shape> batchOutputShapes =
        outputShapes.stream()
            .map(
                shape -> {
                  long[] newShape = new long[shape.numDimensions() + 1];
                  newShape[0] = -1;
                  System.arraycopy(shape.asArray(), 0, newShape, 1, shape.numDimensions());
                  return Shape.of(newShape);
                })
            .collect(Collectors.toList());
```

Reply:
But if the main reason is to add an extra dimension to an existing shape, I would prefer having a method called `Shape prepend(dimensionSizes)` or something like that than a new constructor. 

We should be able also to "reshape" a shape ultimately (i.e. changing [2, 3, 2] to [2, 1, 2, 3]) with a utility called `Shape reshape(dimensionsSizes)`... but it sounds that in your case, `prepend` would be a better pick? Like you did with `head` and `tail`?

Reply:
Makes sense, `prepend()` is pretty nice, I replaced the constructor.